backend/tests/integration/orcid/test_orcid_domain_models.py

```diff
@@ -467,21 +468,39 @@ def test_happy_flow_put(self):
             self.factory.record_metadata.json["dois"] = dois
             self.factory_clashing.record_metadata.json["deleted"] = True
 
-            pusher = domain_models.OrcidPusher(self.orcid, self.recid, self.oauth_token)
-            result_putcode = pusher.push()
-
-        assert self.putcode == result_putcode
-        assert not self.cache.has_work_content_changed(self.inspire_record)
+            with mock.patch(
+                "inspirehep.orcid.tasks.orcid_push.retry"
+            ) as celery_task_with_retry:
+                celery_task_with_retry.side_effect = (
+                    Retry(exc=exceptions.DuplicatedExternalIdentifierPusherException),
+                )
+                with raises(Retry):
+                    tasks.orcid_push.apply_async(
+                        queue="orcid_push_legacy_tokens",
+                        kwargs={
+                            "orcid": self.orcid,
+                            "rec_id": self.recid,
+                            "oauth_token": self.oauth_token,
+                        },
+                        throw=True,
+                    )
 
     def test_duplicated_external_identifier_pusher_exception(self):
-        self.factory_clashing.record_metadata.json["titles"] = [
-            {"source": "submitter", "title": "title1"}
-        ]
         with override_config(
             FEATURE_FLAG_ENABLE_ORCID_PUSH=True,
             FEATURE_FLAG_ORCID_PUSH_WHITELIST_REGEX=".*",
             ORCID_APP_CREDENTIALS={"consumer_key": "0000-0001-8607-8906"},
         ), pytest.raises(exceptions.DuplicatedExternalIdentifierPusherException):
```

Comment:
can you move the `pytest.raises` to the place it actually should raise? so we make sure it doesn't raise too early but exactly where it's expected.